src/common/cockpitpipe.c

```diff
@@ -1116,11 +1116,18 @@ cockpit_pipe_connect (const gchar *name,
   else
     {
       if (!g_unix_set_fd_nonblocking (sock, TRUE, NULL))
-        g_return_val_if_reached (NULL);
+        {
+          close (sock);
+          g_return_val_if_reached (NULL);
```

Comment:
So I looked at ```/usr/include/glib-2.0/glib/gmessages.h```. Because ```g_return_val_if_reached()``` is implemented as a macro, and the macro calls the generic ```g_log``` function instead of ```g_warn_message``` we can't affect the analyzer.